mautic/mautic
# Add schedule emails

| Q                                      | A
| -------------------------------------- | ---
| Bug fix? (use the a.b branch)          | [ ]
| New feature/enhancement? (use the a.x branch)      | [ ]
| Deprecations?                          | [ ]
| BC breaks? (use the c.x branch)        | [ ]
| Automated tests included?              | [ ] 
| Related user documentation PR URL      | mautic/mautic-documentation#... 
| Related developer documentation PR URL | mautic/developer-documentation#... 
| Issue(s) addressed                     | Fixes https://github.com/mautic/mautic/issues/12555  



#### Description:



#### Steps to test this PR:

The PR added for segment emails new schedule options. The replacement for the publish up and publish down options.

Also added new option `Continue sending email to contacts added to the segment after sending has started?` to send emails just until contact's added to segment.

![image](https://github.com/mautic/mautic/assets/462477/cc704436-4ace-49f5-ba76-a43675627029)


 `Continue sending email to contacts added to the segment after sending has started?`

Now you can send emails to the segment to the contacts who were added to the segment before starting sending.

We also add two new states of current status of segment email:

Running
-  If  `Continue sending email to contacts added to the segment after sending has started?` is enabled then email If it's active it's still in state Running
- If  `Continue sending email to contacts added to the segment after sending has started?` is disabled then this state is until  pending counts are 0   (email must be active)

State

- If  `Continue sending email to contacts added to the segment after sending has started?` is disabled then this state is until  pending counts > 0  (email must be active)

Both have different states icon in the list

![image](https://github.com/mautic/mautic/assets/462477/8cb13b73-74dc-4027-a466-240daf54eefc)


And also labels in detail (we tried to make so simple as possible but as clear as possible)

![image](https://github.com/mautic/mautic/assets/462477/1f757f49-18a8-4374-95b3-266a07a0ab69)

![image](https://github.com/mautic/mautic/assets/462477/a52d63c6-601b-47da-b207-23f16b5a4d7a)




1. Open this PR on Gitpod or pull down for testing locally (see docs on testing PRs [here](https://contribute.mautic.org/contributing-to-mautic/tester))
2. Create segment email, see like you are not able to edit publish dates in edit mode
3. Go to detail view
4. Setup schedule new options
5. Tests If works as expecteed
